### PR TITLE
chore: Bump k8s compilation version to 1.29. Also bump all dependencies

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

We need to go throw the breaking change https://github.com/kubernetes/kubernetes/pull/118653, which should not break the API (only the internally used structs)

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: crates/stackable-operator/src/builder/pod/volume.rs [hunk 1]
```diff
@@ -358,9 +358,9 @@ impl SecretOperatorVolumeSourceBuilder {
                 metadata: Some(ObjectMetaBuilder::new().annotations(annotations).build()),
                 spec: PersistentVolumeClaimSpec {
                     storage_class_name: Some("secrets.stackable.tech".to_string()),
-                    resources: Some(ResourceRequirements {
+                    resources: Some(VolumeResourceRequirements {
```
Comment (Techassi) [inline crates/stackable-operator/src/builder/pod/volume.rs:361]: question: This is a non-breaking change for operator users, but a breaking one for library users? The type changes but the (de)serialization is the same, right?

We should mark the changelog entry as breaking in my opinion.

File: crates/stackable-operator/src/builder/pod/volume.rs [hunk 2]
```diff
@@ -358,9 +358,9 @@ impl SecretOperatorVolumeSourceBuilder {
...
                         requests: Some([("storage".to_string(), Quantity("1".to_string()))].into()),
-                        ..ResourceRequirements::default()
+                        ..Default::default()
```
Comment (Techassi) [inline crates/stackable-operator/src/builder/pod/volume.rs:363]: question: I assume the defaults are the same?

File: crates/stackable-operator/src/commons/affinity.rs [hunk 3]
```diff
@@ -85,6 +85,8 @@ pub fn affinity_between_role_pods(
             namespace_selector: None,
             namespaces: None,
             topology_key: TOPOLOGY_KEY_HOSTNAME.to_string(),
+            match_label_keys: None,
```
Comment (Techassi) [inline crates/stackable-operator/src/commons/affinity.rs:88]: question: Were these fields added in K8s `1.29`? Setting them to `None` should be fine then.

File: crates/stackable-operator/src/builder/pod/volume.rs (hunk 1 above)
Comment (sbernauer) [reply crates/stackable-operator/src/builder/pod/volume.rs:361]: I mean technically we (as operator-rs) are not making a breaking change and I think we do not re-export k8s-openapi. But I'll add a changelog entry anyways :)

File: crates/stackable-operator/src/builder/pod/volume.rs (hunk 2 above)
Comment (sbernauer) [reply crates/stackable-operator/src/builder/pod/volume.rs:363]: Yep (everything is None)

File: crates/stackable-operator/src/commons/affinity.rs (hunk 3 above)
Comment (sbernauer) [reply crates/stackable-operator/src/commons/affinity.rs:88]: Yep, https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#matchlabelkeys

**Review by Techassi — APPROVED** [review]

**Pull request merged**